feed processor user agent customization

pmarschik opened the pull request:
Should fix #140

fguillot commented:
There are some errors returned by the linter:

```
$ make lint
/home/travis/gopath/src/miniflux.app/http/client/client.go:36:2: exported var DefaultUserAgent should have comment or be unexported
/home/travis/gopath/src/miniflux.app/http/client/client.go:78:1: exported method Client.WithUserAgent should have comment or be unexported
Found 2 lint suggestions; failing.
```

Also, imports should follow this order: stdlib imports, app imports, and vendor imports.

In your pull-request, you offer the possibility to override to scraper user agent, but I thought the issue was for all HTTP requests made to the website, even when fetching the initial feed?

fguillot commented:
Renaming `ScraperUserAgent` to `UserAgent` is better since we want to use it for feed fetching and scraping.

To fix the original Tumblr issue, the user agent should be also added in the feed creation form in the "advanced options" bloc. That will allow people to override the user agent at the beginning of the process during the feed creation.

Thanks

pmarschik commented:
I've now renamed `ScraperUserAgent` to `UserAgent` (as well as the DB column) and added it to a couple of forms (add and choose subscription) and use it for feed fetching now as well.
The original feed mentioned in #140 now works.

I've also rebased and squashed my previous commits into one.
